cmd/root.go

```diff
@@ -44,52 +47,18 @@ in with avalanche subnet create myNewSubnet.`,
 	rootCmd.PersistentFlags().StringVar(&logLevel, "log-level", "ERROR", "log level for the application")
 
 	// add sub commands
-	rootCmd.AddCommand(subnetCmd)
-	rootCmd.AddCommand(networkCmd)
 
-	// add hidden backend command
-	backendCmd.Hidden = true
-	rootCmd.AddCommand(backendCmd)
-
-	// subnet create
-	subnetCmd.AddCommand(createCmd)
-	createCmd.Flags().StringVar(&filename, "file", "", "file path of genesis to use instead of the wizard")
-	createCmd.Flags().BoolVar(&useSubnetEvm, "evm", false, "use the SubnetEVM as the base template")
-	createCmd.Flags().BoolVar(&useCustom, "custom", false, "use a custom VM template")
-	createCmd.Flags().BoolVarP(&forceCreate, forceFlag, "f", false, "overwrite the existing configuration if one exists")
-
-	// subnet delete
-	subnetCmd.AddCommand(deleteCmd)
-
-	// subnet deploy
-	subnetCmd.AddCommand(deployCmd)
-	deployCmd.Flags().BoolVarP(&deployLocal, "local", "l", false, "deploy to a local network")
-
-	// subnet describe
-	subnetCmd.AddCommand(readCmd)
-	readCmd.Flags().BoolVarP(
-		&printGenesisOnly,
-		"genesis",
-		"g",
-		false,
-		"Print the genesis to the console directly instead of the summary",
-	)
+	// We need to pass double pointers to app into these functions since app
+	// has not been initialized yet
+	subnet := subnetcmd.SetupSubnetCmd(app)
+	rootCmd.AddCommand(subnet)
 
-	// subnet list
-	subnetCmd.AddCommand(listCmd)
+	network := networkcmd.SetupNetworkCmd(app)
+	rootCmd.AddCommand(network)
 
-	// network
-	// network start
-	networkCmd.AddCommand(startCmd)
-
-	// network stop
-	networkCmd.AddCommand(stopCmd)
-
-	// network clean
-	networkCmd.AddCommand(cleanCmd)
-
-	// network status
-	networkCmd.AddCommand(statusCmd)
+	// add hidden backend command
+	backend := backendcmd.SetupBackendCmd(app)
```

Comment:
probably it will be nest to create and setup the command instance in one function, and do eg 
`rootCmd.AddCommand(backendcmd.NewCmd(app))`

Reply:
same for other cobra.Command objects, try to give to each one its own NewCmd() function (if there are many in the same package, call it NewFooCmd, NewBarCmd)

Reply:
This was the way we originally developed it, but we decided to take a more centralized approach. I think this is a decent compromise.